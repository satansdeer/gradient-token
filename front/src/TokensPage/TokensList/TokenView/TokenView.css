.TokenView-image_wrapper {
  background-color: #e7dcd6;
  border-radius: 10px;
  margin-bottom: 20px;
  padding: 50px;
}

.TokenView-label {
  text-align: left;
  margin-bottom: 20px;
}
