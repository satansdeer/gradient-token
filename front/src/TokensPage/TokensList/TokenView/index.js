import React from "react";
import { PropTypes as MobxPropTypes } from "mobx-react";
import TokenImage from "components/TokenImage";
import "./TokenView.css";

const TokenView = ({ token }) => (
  <div>
    <div className="TokenView-image_wrapper">
      <TokenImage size={200} outer={token[0]} inner={token[1]} />
    </div>
    <div className="TokenView-label">{`${token[0]} â€“ ${token[1]}`}</div>
  </div>
);

TokenView.propTypes = {
  token: MobxPropTypes.arrayOrObservableArray
};

export default TokenView;
