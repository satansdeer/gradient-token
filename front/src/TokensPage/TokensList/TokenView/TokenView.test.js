import React from "react";
import renderer from "react-test-renderer";
import TokenView from "./index";

describe("<TokenView />", () => {
  it("renders and matches the snapshot", () => {
    const component = renderer.create(<TokenView token={["#000", "#fff"]} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
