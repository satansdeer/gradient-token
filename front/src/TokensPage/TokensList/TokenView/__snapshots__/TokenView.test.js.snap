// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TokenView /> renders and matches the snapshot 1`] = `
<div>
  <div
    className="TokenView-image_wrapper"
  >
    <svg
      height={200}
      version="1.1"
      width={200}
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <radialGradient
          id="000fff"
        >
          <stop
            offset="0%"
            stopColor="#000"
          />
          <stop
            offset="100%"
            stopColor="#fff"
          />
        </radialGradient>
      </defs>
      <circle
        cx={100}
        cy={100}
        fill="url(#000fff)"
        r={96}
        stroke="#fff"
        strokeWidth="2"
      />
    </svg>
  </div>
  <div
    className="TokenView-label"
  >
    #000 â€“ #fff
  </div>
</div>
`;
